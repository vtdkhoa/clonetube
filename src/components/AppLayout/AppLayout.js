import React from 'react'
import HeaderNav from '../../containers/HeaderNav/HeaderNav'
import ScrollToTop from '../../components/ScrollToTop/ScrollToTop'
import './AppLayout.scss'

const AppLayout = props => {
  return (
    <ScrollToTop>
      <div className="app-layout">
        <HeaderNav/>
        {props.children}
      </div>
    </ScrollToTop>
  )
}

export default AppLayout