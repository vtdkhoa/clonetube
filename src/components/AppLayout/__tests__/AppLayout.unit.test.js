import React from 'react'
import AppLayout from '../AppLayout'
import { shallow } from 'enzyme'

test('renders <AppLayout/> no children', () => {
  const wrapper = shallow(<AppLayout/>)
  expect(wrapper).toMatchSnapshot()
})

test('renders <AppLayout/> with one child', () => {
  const wrapper = shallow(
    <AppLayout>
      <div>Child</div>
    </AppLayout>
  )
  expect(wrapper).toMatchSnapshot()
})

test('renders <AppLayout/> with children', () => {
  const wrapper = shallow(
    <AppLayout>
      <div>Child</div>
      <div>
        <span>Child</span>
        <p>Child</p>
      </div>
    </AppLayout>
  )
  expect(wrapper).toMatchSnapshot()
})