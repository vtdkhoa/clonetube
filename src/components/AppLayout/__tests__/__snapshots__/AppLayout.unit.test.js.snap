// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <AppLayout/> no children 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
</div>
`;

exports[`renders <AppLayout/> with children 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
  <div>
    Child
  </div>
  <div>
    <span>
      Child
    </span>
    <p>
      Child
    </p>
  </div>
</div>
`;

exports[`renders <AppLayout/> with one child 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
  <div>
    Child
  </div>
</div>
`;
